pkg/clusteragent/orchestrator/controller.go

```diff
@@ -55,6 +55,8 @@ type Controller struct {
 	rsListerSync            cache.InformerSynced
 	serviceLister           corelisters.ServiceLister
 	serviceListerSync       cache.InformerSynced
+	nodesLister             corelisters.NodeLister
```

Comment:
we are already using the sharedCache for Nodes somewhere else. From what I can tell we should be fine to use the listener as a new field here.

Reply:
After discussing it offline, the other point will be refactored at an alter point. This will not be part of this PR.